app/controllers/staff/work_histories_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Staff::WorkHistoriesController < ApplicationController
+  layout "staff"
+
+  before_action :require_staff
+  before_action :load_order_detail, only: :edit
+  before_action :load_work_histories, only: :index
+
+  def index; end
+
+  def edit
+    if @order_detail.update_column(:status, OrderDetail.statuses[:finished])
```

Comment:
cập nhật enum e có thể làm như này:
```suggestion
    if @order_detail.finished!
```